File: GoogleCodeJam/StoreCredit.java
import java.util.*;
import java.io.*;

public class StoreCredit
{
    static Scanner in = new Scanner(System.in);
    public static void main(String[] args)
    {
	String line;
	int tc;
	line = in.nextLine();
	tc = Integer.parseInt(line);
	int count = 1;
	while(tc > 0)
	    {
		//System.out.println("count (case): " + count);
		int c; //store credit
		c = Integer.parseInt(in.nextLine());
		//System.out.println("c: "+c);
		int p; //price
		p = Integer.parseInt(in.nextLine());
		//System.out.println("p: " + p);
		int[] array = new int[p];
		int counter = 0;
		while(counter < p)
		    {			
			array[counter] = in.nextInt();
			counter++;
		    }
		boolean found = false;
		counter = 0;
		while(!found || counter == array.length)
		    {
			//System.out.println("counter: " + array[counter]);
			int curr = counter;
			int j = counter+1;
			for(int i=j; i < array.length; i++)
			    {
				//System.out.print("i: " + array[i]);
				//System.out.println("c' :" + (array[curr]+array[i]));
				if(array[curr]+array[i]==c){
				    found = true;
				    int max;
				    int min;
				    if(curr > i)
					{
					    max = curr+1;
					    min = i+1;
					}
				    else
					{
					    max = i+1;
					    min = curr+1;
					}
				    System.out.println("Case #"+count+": "+min+ " " + max);
				    String nothing = in.nextLine();
				}
				//System.out.println();
			    }
			counter++;
		    }
		tc--;
		count++;
	    }
    }
}

File: GoogleCodeJam/storeCredit.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
using namespace std;
#define max(a, b) ((a > b) ? a : b)
#define min(a, b) ((a < b) ? a : b)
int main()
{
	int tc;
	int cn = 1;
	cin >> tc;
	while(tc--)
	{
		int c, p;
		cin >> c; //credit, 5-1000
		cin >> p; //number of prices, 3-2000
		vector<int> prices;
		for(int i = 0; i < p; i++)
		{
			int x;
			cin >> x;
			prices.push_back(x);
		}
		bool found = false;
		cout << "Case #" << cn << ": ";
		for(int i = 0; i < p; i++)
		{
			if(found)	{break;}
			for(int j =0; j < p; j++)
			{
				//cout << prices[i] << " " << prices[j] << endl;
				if(j == i)	{continue;}
				if(prices[i] + prices[j] == c)
				{
					cout << (min(i, j)+1) << " " << (max(i, j)+1);
					found = true;
					break;
				}
			}
		}
		cout << endl;
		cn++;
	}
	return 0;
}
